pandas/io/excel/_base.py

```diff
@@ -1016,28 +1078,30 @@ def __init__(
                     else:
                         stacklevel = 2
                     warnings.warn(
-                        "The xlrd engine is no longer maintained and is not "
-                        "supported when using pandas with python >= 3.9. However, "
-                        "the engine xlrd will continue to be allowed for the "
-                        "indefinite future. Beginning with pandas 1.2.0, the "
-                        "openpyxl engine will be used if it is installed and the "
-                        "engine argument is not specified. Either install openpyxl "
-                        "or specify engine='xlrd' to silence this warning.",
+                        f"Your version of xlrd is {xlrd_version}. In xlrd >= 2.0, "
+                        f"only the xls format is supported. As a result, the "
+                        f"openpyxl engine will be used if it is installed "
+                        f"and the engine argument is not specified. Either install "
+                        f"openpyxl or specify engine='xlrd' to silence this warning.",
```

Comment:
So we should probably not silence the warning when `engine="xlrd"` is specified? Since in the future we want to disallow any reading of non-xls files with xlrd?

Reply:
Yea - agreed. I reworked the logic so whenever engine="xlrd" and xlrd is installed we either raise a ValueError or FutureWarning.